### ISSUE TITLE
Import order is not preserved when building from ES6 modules.

### ISSUE THREAD

**Issue by bmomberger-bitovi** [stealjs/steal-tools#804]
```javascript
// app.js
import 'polyfills';
import thing from 'code/where/polyfills/needed';
```

when built becomes 

```javascript
define("myapp@1.0.0#app.js", [
  'exports'
  'code/where/polyfills/needed',
  'polyfills',
  function(_needed) {
    var _needed2 = _interopRequireDefault(_needed);
...
})
```

Thus, the polyfills are imported after where they're needed, causing breakage that only shows up when running the built production bundle.  The intended order would be preserved here if running the app using a development-mode steal.

Recommend adding nonce names in the function signature instead of reordering dependencies, i.e. ```$$__steal_module_polyfills```

**Comment by matthewp** [comment]
I don't think we reorder dependencies, we probably just don't have any code that attempts to preserve the order.  My guess is that this is *probably* babel's AMD transformer. @m-mujica could you try with babel (maybe just with the cli tool) and see if the order is preserved, or if we are doing something that causes the order to change?

**Comment by m-mujica** [comment]
The reorder is indeed introduced by babel's amd transform

```js
// index.js
import "./a";
import b from "./b";

export default function() {
  return a + b;
}
```

Running babel using the CLI 

```
./node_modules/babel-cli/bin/babel.js --plugins transform-es2015-modules-amd index.js
```

I got:

```js
define(["exports", "./b", "./a"], function (exports, _b) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  exports.default = function () {
    return a + _b2.default;
  };

  var _b2 = _interopRequireDefault(_b);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }
});
```

**Comment by matthewp** [comment]
What about the commonjs transform? We could possible use it instead, and then do cjs->amd transpiling ourselves.

**Comment by m-mujica** [comment]
That's the CJS output:

```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.default = function () {
  return a + _b2.default;
};

require("./a");

var _b = require("./b");

var _b2 = _interopRequireDefault(_b);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
```

There is no need to change the import order in CJS, I mean, they do it with the AMD output to avoid having null arguments in the define factory function *I guess*.

I'm going to try getting babel -> cjs -> amd to work.

**Comment by matthewp** [comment]
@bmomberger-bitovi @mjstahl so we've confirmed that it's a Babel bug. We're filing a bug there. In the meantime you can workaround this by configuring `'code/where/polyfills/needed'` to depend on `'polyfills'` so that the execution order will be guaranteed.  If it's your own module you can add the import to it, if not you can use meta config:

```json
"steal": {
  "meta": {
    "code/where/polyfills/needed": {
      "deps": ["polyfills"]
    }
  }
}
```

**Comment by m-mujica** [comment]
fwiw the issue was already reported to babel https://github.com/babel/babel/pull/5474

But it has not made it yet to babel-standalone which is the package that steal-tool uses, once `babel-standalone@7.x` is released we can just upgrade to that version and close this issue.

**Comment by bmomberger-bitovi** [comment]
Thanks for all the good detective work on this.  For now we've worked around it by using `import foo from 'bar';` style, with an eslint exception, for 'polyfills'.
